lib/Cake/Test/Case/Model/CakeSchemaTest.php

```diff
@@ -650,6 +683,34 @@ public function testSchemaReadWithCrossDatabase() {
 	}
 
 /**
+ * testSchemaRead method when a primary key is on a non-conventional column
+ *
+ * @return void
+ */
+	public function testSchemaReadWithNonConventionalPrimaryKey() {
+		$this->skipIf($this->db instanceof Postgres, 'Cannot test on Postgres');
+		$db = ConnectionManager::getDataSource('test');
+		$fixture = new NonConventionalPrimaryKeyFixture();
+		$fixture->create($db);
+
+		$read = $this->Schema->read(array(
+			'connection' => 'test',
+			'name' => 'TestApp',
+			'models' => false
+		));
+		$fixture->drop($db);
+
+		$hasTable = isset($read['tables']['non_conventional']);
+		$this->assertTrue($hasTable, 'non_conventional table should appear');
+		$versionIdHasKey = $hasTable && isset($read['tables']['non_conventional']['version_id']['key']);
```

Comment:
Sure... as long as nothing happens after the assertions, or your test never fails. As it turns out I based my test on the one above, `testSchemaReadWithCrossDatabase()`. In this test, `$fixture->drop($db);` is placed *after* all the assertions. If any assertion fails, the fixture is not dropped and sticks around, and you won't be able to run the test anymore, you will get a `Error: Fixture creation for "non_conventional" failed "SQLSTATE[42S01]: Base table or view already exists: 1050 Table 'non_conventional' already exists"` each time,  until you manually clean the database. That's why I added this `$hasTable` extra test... until I decided to move `$fixture->drop($db);` above all the assertions anyway -- so you are correct they are not needed anymore and I'll remove them. But the test above definitely has that problem...